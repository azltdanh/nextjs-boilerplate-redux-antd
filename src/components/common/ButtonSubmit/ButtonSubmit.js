import { Button } from 'antd';

const ButtonSubmit = ({ className, loading, children }) => (
  <Button
    size="large"
    type="primary"
    htmlType="submit"
    block
    loading={loading}
    disabled={loading}
    className={className}
  >
    {children}
  </Button>
);

export default ButtonSubmit;
