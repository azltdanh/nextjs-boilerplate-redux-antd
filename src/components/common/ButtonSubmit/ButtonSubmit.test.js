import ButtonSubmit from './ButtonSubmit';

describe(ButtonSubmit.name, () => {
  it('renders', () => {
    const wrapper = shallow(<ButtonSubmit>Submit</ButtonSubmit>);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with props', () => {
    const wrapper = shallow(
      <ButtonSubmit className="css-class" loading>
        Submit
      </ButtonSubmit>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
