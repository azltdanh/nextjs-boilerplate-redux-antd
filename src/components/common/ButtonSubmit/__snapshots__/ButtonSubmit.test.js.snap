// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonSubmit renders 1`] = `
<Button
  block={true}
  ghost={false}
  htmlType="submit"
  loading={false}
  size="large"
  type="primary"
>
  Submit
</Button>
`;

exports[`ButtonSubmit renders with props 1`] = `
<Button
  block={true}
  className="css-class"
  disabled={true}
  ghost={false}
  htmlType="submit"
  loading={true}
  size="large"
  type="primary"
>
  Submit
</Button>
`;
